components/coding/profileV2/achievement_components/AchievementComponent.tsx

```diff
@@ -54,31 +57,39 @@ const AchievementComponent = ({
       userId
     ).removedBadges;
 
-    if (addedBadges.length > 0) {
-      saveAddedBadges({
-        variables: {
-          objects: addedBadges,
-        },
-      });
-    }
-    if (removedBadges.length > 0) {
-      removedBadges.forEach((badge) => {
-        saveRemovedBadges({
-          variables: {
-            badgeId: badge.badgeId,
-            userId: badge.userId,
-          },
-        });
-      });
-    }
-    alert("Your badge selections have been updated.");
+    Promise.all([
```

Comment:
Very cool to see you using promises in such a complex way, awesome job. Lets chat about promises this week and I'll assess your knowledge of them. If you understand promises well, I'll create an advanced JS badge for you on skillify and issue it to you.